### PR TITLE
Event register

### PR DESCRIPTION
**Pull request by fjwillemsen**
Registration process for a new user and the event

### CODE REVIEW

File: GoedeDoelenHelpen/ClientApp/src/app/event/default-event-register/default-event-register.component.ts [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+  validPassword(password: AbstractControl) {
+    return new RegExp('(?=.*[a-zA-Z])(?=.*[0-9])(?=.*[A-Z])').test(password.value);
+  }
+
+  equalPasswords(password: AbstractControl, repeat: AbstractControl): boolean {
+    return password.value === repeat.value ? true : false;
+  }
+
+  allRequiredFieldsValid() {
+    if (
```
Comment (barld) [inline GoedeDoelenHelpen/ClientApp/src/app/event/default-event-register/default-event-register.component.ts]: you could als create a group of groups so you can use the built in api of FormGroup https://angular.io/api/forms/FormGroup

File: GoedeDoelenHelpen/Startup.cs [hunk 2]
```diff
@@ -49,7 +49,6 @@ public void ConfigureServices(IServiceCollection services)
                 options.Password.RequireNonAlphanumeric = false;
                 options.Password.RequireUppercase = true;
                 options.Password.RequireLowercase = false;
-                options.Password.RequiredUniqueChars = 6;
```
Comment (barld) [inline GoedeDoelenHelpen/Startup.cs:77]: complete away?

File: appveyor.yml [hunk 3]
```diff
@@ -13,32 +14,29 @@ install:
...
+  publish_wap_xcopy: true
+  verbosity: normal
 build_script:
-  # output will be in ./GoedeDoelenHelpen/bin/debug/netcoreapp2.1/publish
+  # output will be in ./GoedeDoelenHelpen/bin/x86/Debug/netcoreapp2.1/publish
   - cmd: dotnet publish ./GoedeDoelenHelpen/GoedeDoelenHelpen.csproj
 after_build:
   - ps: cd ./GoedeDoelenHelpen/ClientApp
-  - ps: npm run build-ci
+  - ps: npm run build-ci --prod
```
Comment (barld) [inline appveyor.yml]: `--prod` is doing nothing as far I know

**Review by barld — COMMENTED** [review]
there is also a merge conflict

File: GoedeDoelenHelpen/Startup.cs (hunk 2 above)
Comment (fjwillemsen) [reply GoedeDoelenHelpen/Startup.cs:77]: We already have at least 2 unique chars with a required upper case and required number

**Review by barld — DISMISSED** [review]

**Review by barld — DISMISSED** [review]

**Review by barld — APPROVED** [review]

**Pull request merged**